azurelinuxagent/ga/update.py

```diff
@@ -421,6 +422,28 @@ def _emit_restart_event(self):
 
         return
 
+    @staticmethod
+    def _emit_changes_in_default_configuration():
```

Comment:
Thanks for adding this, much needed! 
I have a suggestion, why not just log all parameters that changed from default, that way we will always have an idea of what the customers are using and maybe later on we can leverage that to change our defaults if needed.